Formula/qt@5.rb

```diff
@@ -75,6 +75,16 @@ class QtAT5 < Formula
     depends_on "zstd"
   end
 
+  # Patch for qmake on ARM
+  # https://codereview.qt-project.org/c/qt/qtbase/+/327649
+  on_arm do
+    patch do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/9dc732/qt/qt-split-arch.patch"
+      sha256 "36915fde68093af9a147d76f88a4e205b789eec38c0c6f422c21ae1e576d45c0"
+      directory "qtbase"
+    end
+  end
```

Comment:
This is going here because there can only be one of each `on_{system}` block per formula (because that was the rule with `on_macos` and `on_linux`

Reply:
Hmm... maybe? I can't find an example off the top of my head

Although I'd add that the rule didn't apply across the main body and resources meaning there could be an `on_linux` around a `depends_on` line and also one inside a `resource` adding a patch to that resource.

Like:

```ruby
class Foo < Formula
  ...

  on_linux do
    depends_on "foo"
  end

  resource do
    URL "..."
    sha256 ".."

    on_linux do
      patch do
        ...
      end
    end
  end
end

Reply:
I think if you reverse it like
```ruby
patch do
  on_arm do
    url "..."
    sha256 "..."
  end
end
```
then it should work. Maybe.

Reply:
Yep, great idea!

Reply:
I do like this idea, but it doesn't look like it will work currently since `patch` without any info isn't allowed. We should discuss this in https://github.com/Homebrew/brew/pull/13539